client/src/components/pages/RobotInteractionCockpit/RobotInteractionCockpit.jsx

```diff
@@ -0,0 +1,150 @@
+
+const { Step } = Steps;
+const { Title } = Typography;
+
+/**
+ * @description Page, where you can interact with a robot and for example enter input
+ * @category Client
+ * @component
+ */
+const RobotInteractionCockpit = (match) => {
+  const { robotId } = match.match.params;
+  const { userId } = match.location.state;
+  const isMounted = useRef(true);
+  const [parameterList, setParameterList] = useState([]);
+  const [currentStep, setCurrentStep] = useState(0);
+
+  /**
+   * @description For each activity of the current robot get the id, the name and all the parameters that require a user input
+   */
+  const getActivityAndParameterInformation = useCallback(() => {
+    getAllParametersForRobot(robotId)
+      .then((response) => response.json())
+      .then((parameterObjects) => {
+        const activityInformationList = [];
+        Array.prototype.forEach.call(parameterObjects, (parameterObject) => {
+          const { activityId } = parameterObject;
+          let activityName = '';
+          let ssot = sessionStorage.getItem('ssotLocal');
+          ssot = JSON.parse(ssot);
+          Array.prototype.forEach.call(ssot.elements, (elem) => {
+            if (elem.id === activityId) {
+              activityName = elem.name;
+            }
+          });
+          const activityParameter = [];
+          Array.prototype.forEach.call(
+            parameterObject.rpaParameters,
+            (parameter) => {
+              if (parameter.requireUserInput) {
+                activityParameter.push(parameter);
+              }
+            }
+          );
+          if (activityParameter.length !== 0) {
+            const activityInformation = {
+              activityId,
+              activityParameter,
+              activityName,
+            };
+            activityInformationList.push(activityInformation);
+          }
+        });
+        if (isMounted.current) {
+          setParameterList(activityInformationList);
+        }
+      });
+  }, [robotId]);
+
+  /**
+   * @description Equivalent to ComponentDidMount in class based components
+   */
+  useEffect(
+    () => () => {
+      isMounted.current = true;
+    },
+    []
+  );
+
+  /**
+   * @description Ensures that we only update the paramterList when the paramters we fetch form the DB have changed
+   */
+  useEffect(() => {
+    getActivityAndParameterInformation();
+  }, [getActivityAndParameterInformation]);
+
+  /**
+   * @description Sends a job to the server to execute a specfic robot for a specific user
+   */
+  const startRobot = async () => {
+    const robotIsExecutable = await isRobotExecutable(robotId);
+    if (robotIsExecutable) {
+      upsert().then(() => {
+        setCurrentStep(1);
+        socket.emit('robotExecutionJobs', { robotId, userId });
+      });
+    } else {
+      alert('Your Bot is not fully configured and can not be executed!');
+    }
+  };
+
+  return (
+    <Layout>
+      <HeaderNavbar selectedKey={4} />
```

Comment:
At least in my testing this was not correctly displayed, as there is no 4th tab in the header navbar. From my point it should either display that additional tab, or only display the logo, so no editing tabs at all, which I would prefer

Reply:
Thanks for the hint. I agree. I changed the header nav bar now so that only the tab for the robot interaction cockpit is displayed when we are on the RobotInteractionCockpit page. ✅

Reply:
update: I also put the robot overview tab back in the header bar so that we can easily navigate back to the overview page